2-types/part_1-2/src/part_2.rs

```diff
@@ -1,49 +1,83 @@
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
 
-// Part 2: Safer time handling with Option<Result> and pattern matching.
-// We model a device with optional local_time parsed from string.
-// We expose a method to render local_time, or return a descriptive error.
+use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone)]
-pub struct Device {
-    pub id: String,
-    pub local_time: Option<String>,
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Request {
+    pub r#type: String,
+    pub stream: Stream,
+    pub gifts: Vec<Gift>,
+    pub debug: DebugInfo,
 }
 
-#[derive(Debug, thiserror::Error)]
-pub enum TimeError {
-    #[error("local time is missing")]
-    Missing,
-    #[error("invalid local time format: {0}")]
-    Invalid(String),
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Stream {
+    pub user_id: String,
+    pub is_private: bool,
+    pub settings: i64,
+    pub shard_url: String,
+    pub public_tariff: Tariff,
+    pub private_tariff: Tariff,
 }
 
-impl Device {
-    pub fn parsed_local_time_minutes(&self) -> Result<i64, TimeError> {
-        let Some(ref s) = self.local_time else { return Err(TimeError::Missing) };
-        let parts: Vec<&str> = s.split(|c| c==' ' || c==':' || c=='-').collect();
-        if parts.len() < 5 { return Err(TimeError::Invalid(s.clone())); }
-        let hh = parts[3].parse::<i64>().map_err(|_| TimeError::Invalid(s.clone()))?;
-        let mm = parts[4].parse::<i64>().map_err(|_| TimeError::Invalid(s.clone()))?;
-        Ok(hh*60 + mm)
-    }
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Tariff {
+    pub id: Option<i64>,           
+    pub client_price: Option<i64>, 
+    pub price: Option<i64>,
+    pub duration: String,
+    pub description: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Gift {
+    pub id: i64,
+    pub price: i64,
+    pub description: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DebugInfo {
+    pub duration: String,
+    pub at: String,
```

Comment:
use either `OffsetDateTime`/`UtcDateTime` from the `time` crate or `DateTime` from the `chrono` crate